cmd/kubeadm/app/apis/kubeadm/v1beta3/types.go

```diff
@@ -128,6 +128,14 @@ type ClusterConfiguration struct {
 	// +optional
 	ImageRepository string `json:"imageRepository,omitempty"`
 
+	// ImageRepositoryUsername username for image repository if any
+	// +optional
+	ImageRepositoryUsername string
+
+	// ImageRepositoryPassword password for image repository if any
+	// +optional
+	ImageRepositoryPassword string
```

Comment:
v1beta2 and v1beta3 are locked and new fields cannot be added, thus such a change must be in a future release.
kubeadm APIs are different in this regard to the rest of k8s.